doc/developer/design/20220314_linearized_reads.md

```diff
@@ -0,0 +1,71 @@
+in some of academia. This is effectively _serializability_ + _linearizability_, but because we consider _serializability_ important
+by default, we refer to the new concept we are introducing.
+
+See [this jepsen map on consistency](https://jepsen.io/consistency) for more details about how these terms are used.
+
+Also, a previous version of this document included a design that has since been [broken out](https://github.com/MaterializeInc/materialize/pull/11302).
+
+## Summary
+
+Materialize has "real time" sources whose updates are timestamped with a `materialized` local timestamp, used for queries and interaction with other data.
+These timestamps are produced as _timestamp bindings_, which are durably-stored associations between `timestamps` (the literal, real-time timestamps that mz uses to drive its dataflows) and `offsets` (consistent, ordered, source-specific integer identifiers that express how _far along_ you are in the source).
+
+These bindings are produced and durably retained for sources, though the behavior depends on whether the source is currently
+
+- "materialized", in that it is actively being consumed in a live dataflow. This process will continually pull data and create bindings.
+
+- "unmaterialized", in that the source is known but not actively consumed. No current process will create new bindings.
+
+The timestamp bindings are the basis of **consistency** in Materialize. By reading from sources, and views over the sources, at a single timestamp, we can ensure that all readers observe consistent results, for potenially varying definitions of "consistent".
+
+How we choose this timestamp influences the type of consistency guarantee we get. In particular, if we want to achieve _linearizable_ reads, those that reflect all data available in upstream sources as of at least the moment `SELECT` was typed, we will need to select a timestamp that whose bindings reflect all available source data. Consequently, STORAGE will need a command to provide such a timestamp for a collection of sources, when prompted.
+
+## Goals
+
+Offer _linearizable_ reads in as many cases as possible. `materialized` is ultimately built to support this!
+
+Avoid performance interference of linearizable reads on other "less consistent" reads, such as sequential or eventual consistency.
+
+## Non-Goals
+
+Alter the default experience of using `materialized` (yet).
+
+Present strong opinions on which consistency levels should be the default.
+
+## Description
+
+### Background
+
+Sources currently consume messages that contain offsets, and produce updates that contain timestamps. Those offsets are written down, associated with a timestamp, as a _timestamp binding_.
+When you do a peek (read) against some source. A recent timestamp is chosen, the index is consulted for data that is valid in respect to that timestamp
+and the _timestamp bindings_.
+
+### Linearizable reads
+
+In the case of a linearizable read, peeks will be augemented to work as follows
```

Comment:
I'm also interested in a description of exactly what will change. This would clarify things because we already do things that look mostly like this proposal. For example we do consult storage for its current contents (https://github.com/MaterializeInc/materialize/blob/9bfed211f52f1cdac27a7cc7d17d1107f33d4975/src/coord/src/coord.rs#L3453) and that time contributes to the peek time.